File: VectoCore/VectoCoreTest/Utils/MockVehicleInputData.cs
using System;
using System.Collections.Generic;
using System.Xml;
using TUGraz.VectoCommon.BusAuxiliaries;
using TUGraz.VectoCommon.InputData;
using TUGraz.VectoCommon.Models;
using TUGraz.VectoCommon.Utils;

namespace TUGraz.VectoCore.Tests.Utils
{
	public class MockVehicleInputData : IVehicleDeclarationInputData
	{
		public DataSource DataSource { get; set; }

		public bool SavedInDeclarationMode { get; set; }

		public string Manufacturer { get; set; }

		public string Model { get; set; }

		public DateTime Date { get; set; }

		public string SimulationToolLicenseNumber { get; set; }

        public Kilogram H2StorageUsableCapacity { get; set; }

        public HydrogenStorageTechnology? HydrogenStorageTechnology { get; set; }

        public bool BatteryOnlyMode { get; set; }
       
		public DynamicChargingTechnology DynamicChargingTechnology { get; set; }

        public string AppVersion { get; }

		public CertificationMethod CertificationMethod { get; set; }

		public string CertificationNumber { get; set; }

		public DigestData DigestValue { get; set; }

		public string Identifier { get; set; }

		public bool ExemptedVehicle { get; set; }

		public string VIN { get; set; }

		public TUGraz.VectoCommon.Models.LegislativeClass? LegislativeClass { get; set; }

		public VehicleCategory VehicleCategory { get; set; }

		public AxleConfiguration AxleConfiguration { get; set; }

		public Kilogram CurbMassChassis { get; set; }

		public Kilogram GrossVehicleMassRating { get; set; }

		public IList<ITorqueLimitInputData> TorqueLimits { get; set; }

		public string ManufacturerAddress { get; set; }

		public PerSecond EngineIdleSpeed { get; set; }

		public bool VocationalVehicle { get; set; }

		public bool? SleeperCab { get; set; }

		public bool? AirdragModifiedMultistep { get; }

		public TUGraz.VectoCommon.InputData.TankSystem? TankSystem { get; set; }

		public IAdvancedDriverAssistantSystemDeclarationInputData ADAS { get; set; }
		public IVehicleInMotionChargingDeclaration InMotionCharging { get; }

		public bool ZeroEmissionVehicle { get; set; }

		public bool HybridElectricHDV { get; set; }

		public bool DualFuelVehicle { get; set; }

		public Watt MaxNetPower1 { get; set; }


		public string ExemptedTechnology { get; }

		public RegistrationClass? RegisteredClass { get; }

		public int? NumberPassengerSeatsUpperDeck { get; }

		public int? NumberPassengerSeatsLowerDeck { get; }

		public int? NumberPassengersStandingLowerDeck { get; }

		public int? NumberPassengersStandingUpperDeck { get; }

		public CubicMeter CargoVolume { get; }

		public TUGraz.VectoCommon.Models.VehicleCode? VehicleCode { get; }

		public bool? LowEntry { get; }

		public bool Articulated { get; }

		public Meter Height { get; }

		public Meter Length { get; }

		public Meter Width { get; }

		public Meter EntranceHeight { get; }

		public ConsumerTechnology? DoorDriveTechnology { get; }

		public VehicleDeclarationType VehicleDeclarationType { get; }

		public IDictionary<PowertrainPosition, IList<Tuple<Volt, TableData>>> ElectricMotorTorqueLimits { get; }

		public TableData BoostingLimitations => throw new NotImplementedException();

		public IVehicleComponentsDeclaration Components { get; set; }

		public XmlNode XMLSource { get; }

		public string VehicleTypeApprovalNumber { get; }

		public ArchitectureID ArchitectureID { get; }

		public bool OVC { get; }

		public Watt MaxChargingPower { get; }
		public VectoSimulationJobType VehicleType { get; }
	}
}

File: VECTO/Input Files/MockVehicleInputData.vb
Imports System.Collections.Generic
Imports System.Xml
Imports TUGraz.VectoCommon.BusAuxiliaries
Imports TUGraz.VectoCommon.InputData
Imports TUGraz.VectoCommon.Models
Imports TUGraz.VectoCommon.Utils

Public Class MockVehicleInputData
    Implements IVehicleDeclarationInputData
    Public Property DataSource As DataSource Implements IComponentInputData.DataSource
    Public Property SavedInDeclarationMode As Boolean Implements IComponentInputData.SavedInDeclarationMode
    Public Property Manufacturer As String Implements IComponentInputData.Manufacturer
    Public Property Model As String Implements IComponentInputData.Model
    Public Property [Date] As DateTime Implements IComponentInputData.[Date]
    Public ReadOnly Property AppVersion As String Implements IComponentInputData.AppVersion
    Public Property CertificationMethod As CertificationMethod Implements IComponentInputData.CertificationMethod
    Public Property CertificationNumber As String Implements IComponentInputData.CertificationNumber
    Public Property DigestValue As DigestData Implements IComponentInputData.DigestValue
    Public Property Identifier As String Implements IVehicleDeclarationInputData.Identifier
    Public Property ExemptedVehicle As Boolean Implements IVehicleDeclarationInputData.ExemptedVehicle
    Public Property VIN As String Implements IVehicleDeclarationInputData.VIN
    Public Property LegislativeClass As LegislativeClass? Implements IVehicleDeclarationInputData.LegislativeClass
    Public Property VehicleCategory As VehicleCategory Implements IVehicleDeclarationInputData.VehicleCategory
    Public Property AxleConfiguration As AxleConfiguration Implements IVehicleDeclarationInputData.AxleConfiguration
    Public Property CurbMassChassis As Kilogram Implements IVehicleDeclarationInputData.CurbMassChassis
    Public Property GrossVehicleMassRating As Kilogram Implements IVehicleDeclarationInputData.GrossVehicleMassRating
    Public Property TorqueLimits As IList(Of ITorqueLimitInputData) Implements IVehicleDeclarationInputData.TorqueLimits
    Public Property ManufacturerAddress As String Implements IVehicleDeclarationInputData.ManufacturerAddress
    Public Property EngineIdleSpeed As PerSecond Implements IVehicleDeclarationInputData.EngineIdleSpeed
    Public Property VocationalVehicle As Boolean Implements IVehicleDeclarationInputData.VocationalVehicle
    Public Property SleeperCab As Boolean? Implements IVehicleDeclarationInputData.SleeperCab
    Public ReadOnly Property AirdragModifiedMultistep As Boolean? Implements IVehicleDeclarationInputData.AirdragModifiedMultistep
    Public Property TankSystem As TankSystem? Implements IVehicleDeclarationInputData.TankSystem

    Public Property ADAS As IAdvancedDriverAssistantSystemDeclarationInputData _
        Implements IVehicleDeclarationInputData.ADAS

    Public ReadOnly Property InMotionCharging As IVehicleInMotionChargingDeclaration Implements IVehicleDeclarationInputData.InMotionCharging

    Public Property ZeroEmissionVehicle As Boolean Implements IVehicleDeclarationInputData.ZeroEmissionVehicle
    Public Property HybridElectricHDV As Boolean Implements IVehicleDeclarationInputData.HybridElectricHDV
    Public Property DualFuelVehicle As Boolean Implements IVehicleDeclarationInputData.DualFuelVehicle
    Public Property MaxNetPower1 As Watt Implements IVehicleDeclarationInputData.MaxNetPower1
    Public ReadOnly Property ExemptedTechnology As String Implements IVehicleDeclarationInputData.ExemptedTechnology
    Public ReadOnly Property RegisteredClass As RegistrationClass? Implements IVehicleDeclarationInputData.RegisteredClass
    Public ReadOnly Property NumberPassengerSeatsUpperDeck As Integer? Implements IVehicleDeclarationInputData.NumberPassengerSeatsUpperDeck
    Public ReadOnly Property NumberPassengerSeatsLowerDeck As Integer? Implements IVehicleDeclarationInputData.NumberPassengerSeatsLowerDeck
    Public ReadOnly Property NumberPassengersStandingLowerDeck As Integer? Implements IVehicleDeclarationInputData.NumberPassengersStandingLowerDeck
    Public ReadOnly Property NumberPassengersStandingUpperDeck As Integer? Implements IVehicleDeclarationInputData.NumberPassengersStandingUpperDeck
    Public ReadOnly Property CargoVolume As CubicMeter Implements IVehicleDeclarationInputData.CargoVolume
    Public ReadOnly Property VehicleCode As VehicleCode? Implements IVehicleDeclarationInputData.VehicleCode
    Public ReadOnly Property LowEntry As Boolean? Implements IVehicleDeclarationInputData.LowEntry
    Public ReadOnly Property Articulated As Boolean Implements IVehicleDeclarationInputData.Articulated
    Public ReadOnly Property Height As Meter Implements IVehicleDeclarationInputData.Height
    Public ReadOnly Property Length As Meter Implements IVehicleDeclarationInputData.Length
    Public ReadOnly Property Width As Meter Implements IVehicleDeclarationInputData.Width
    Public ReadOnly Property EntranceHeight As Meter Implements IVehicleDeclarationInputData.EntranceHeight
    Public ReadOnly Property DoorDriveTechnology As ConsumerTechnology? Implements IVehicleDeclarationInputData.DoorDriveTechnology
    Public ReadOnly Property VehicleDeclarationType As VehicleDeclarationType Implements IVehicleDeclarationInputData.VehicleDeclarationType
    Public ReadOnly Property ElectricMotorTorqueLimits As IDictionary(Of PowertrainPosition, IList(Of Tuple(Of Volt, TableData))) Implements IVehicleDeclarationInputData.ElectricMotorTorqueLimits
    Public ReadOnly Property BoostingLimitations As TableData Implements IVehicleDeclarationInputData.BoostingLimitations
        Get
            Throw New NotImplementedException
        End Get
    End Property

    Public Property Components As IVehicleComponentsDeclaration Implements IVehicleDeclarationInputData.Components
    Public ReadOnly Property XMLSource As XmlNode Implements IVehicleDeclarationInputData.XMLSource
    Public ReadOnly Property VehicleTypeApprovalNumber As String Implements IVehicleDeclarationInputData.VehicleTypeApprovalNumber
    Public ReadOnly Property ArchitectureID As ArchitectureID Implements IVehicleDeclarationInputData.ArchitectureID
    Public ReadOnly Property OvcHev As Boolean Implements IVehicleDeclarationInputData.OvcHev
    Public ReadOnly Property MaxChargingPower As Watt Implements IVehicleDeclarationInputData.MaxChargingPower
    Public ReadOnly Property VehicleType As VectoSimulationJobType Implements IVehicleDeclarationInputData.VehicleType
End Class